Classes/IndexQueue/RecordMonitor.php

```diff
@@ -300,36 +125,25 @@ protected function applyRecordChangesToQueue($table, $uid, $pid)
      * @param DataHandler $tceMain TYPO3 Core Engine parent object
      * @return void
      */
-    public function processDatamap_afterDatabaseOperations($status, $table, $uid, array $fields, DataHandler $tceMain) {
-        $recordTable = $table;
+    public function processDatamap_afterDatabaseOperations($status, $table, $uid, array $fields, DataHandler $tceMain): void
+    {
         $recordUid = $uid;
-
+        $table = (string)$table;
         if ($this->skipMonitoringOfTable($table)) {
             return;
         }
 
         if ($status === 'new' && !MathUtility::canBeInterpretedAsInteger($recordUid)) {
             $recordUid = $tceMain->substNEWwithIDs[$recordUid];
         }
-        if ($this->isDraftRecord($table, $recordUid)) {
+        if (Util::isDraftRecord($table, (int)$recordUid)) {
```

Comment:
We should avoid static methods and Util classes.
See https://phpunit.readthedocs.io/en/9.5/test-doubles.html

Reply:
I agree in principle, we should reconsider the concept of the Util class, especially with regard to the [TYPO3 coding guidelines](https://docs.typo3.org/m/typo3/reference-coreapi/11.5/en-us/CodingGuidelines/CglPhp/PhpArchitecture/ModelingCrossCuttingConcerns/StaticMethods.html) and notes on static methods and utitily classes. 

The old class method `isDraftRecord`  was only a wrapper for Util, but since the calls in our unit tests are unproblematic at this point and it is also considered bad practice to test against mocks, the reintroduction of the wrapper method wouldn't really help us in my view.